backend/drm/drm.c

```diff
@@ -1304,8 +1304,17 @@ void scan_drm_connectors(struct wlr_drm_backend *drm) {
 				wl_list_insert(&wlr_conn->output.modes, &mode->wlr_mode.link);
 			}
 
+			size_t path_len;
+			bool is_mst = false;
+			char *path = get_drm_prop_blob(drm->fd, wlr_conn->id,
+				wlr_conn->props.path, &path_len);
+			if(path_len > 4 && path && strncmp(path, "mst:", 4) == 0) {
+				is_mst = true;
+				free(path);
```

Comment:
Put the free outside of the if statement. For example, if PATH was ever to change and not include `mst:`, this would be a memory leak.
`free(NULL);` is a no-op, so you don't need to worry about `get_drm_prop_blob` failing.

Reply:
Absolutely, amended.